src/asr/asr_utils.py

```diff
@@ -10,19 +10,21 @@
 import chainer
 from chainer import training
 
+# io related
+import kaldi_io_py
 
 # * -------------------- training iterator related -------------------- *
 def make_batchset(data, batch_size, max_length_in, max_length_out, num_batches=0):
     # sort it by input lengths (long to short)
     sorted_data = sorted(data.items(), key=lambda data: int(
-        data[1]['ilen']), reverse=True)
```

Comment:
Same comments with the options. For now, it's better to add error messages (ask the user to redo the "stage 2: Dictionary and Json Data Preparation") instead of removing these four options. Later, we can remove these options.
You don't have to add the error message everywhere. Only this part should be fine.

Reply:
I have a question.
add error messages to asr_utils.py?
not asr_recog.py?

Reply:
`asr_recog.py` will be called with your new data preparation, and we would not require the error message. Some confusion will be happened in the training, and I think the message in options could be sufficient, but if we have the error message, it would be further helpful for users.

Reply:
I am assuming that many people usually skip the "stage 2: Dictionary and Json Data Preparation", since it requires to be called only once. Then, if they pull this change, and perform their experiments with `--stage 3`, the code does not work and it causes some confusions. With the error message in my suggestion, we can avoid confusions for this case.

Reply:
I agree with your idea, i.e. we should add error messages in asr_train.py.
However, I cannnot grasp your comment.
What is the options in asr_utils.py?

Do you mean that, in make_batchset, we should add error messages when data[1] does not have 'input' or 'output' keys?